#
# Submitted-By: Robert Schwebel, 2006-05-26
# Committed-By: Robert Schwebel
#
# Error:
#
# variables.c:3674: error: 'sv_histchars' undeclared here (not in a function)
#
# Description:
#
# In bashhist.c, sv_histchars is used when HISTORY is set, but not
# within an #ifdef BANG_HISTORY section. So we modify it to be defined
# for HISTORY || BANG_HISTORY.
#
# State:
# 
# unknown
#

diff -urN bash-3.0-orig/variables.c bash-3.0/variables.c
--- bash-3.0-orig/variables.c	2004-07-04 19:57:26.000000000 +0200
+++ bash-3.0/variables.c	2006-05-26 08:55:35.000000000 +0200
@@ -3670,9 +3670,9 @@
   { "TZ", sv_tz },
 #endif
 
-#if defined (HISTORY) && defined (BANG_HISTORY)
+#if defined (HISTORY) || defined (BANG_HISTORY)
   { "histchars", sv_histchars },
-#endif /* HISTORY && BANG_HISTORY */
+#endif /* HISTORY || BANG_HISTORY */
 
   { "ignoreeof", sv_ignoreeof },
 
@@ -3885,7 +3885,7 @@
     }
 }
 
-#if defined (BANG_HISTORY)
+#if defined(HISTORY) || defined (BANG_HISTORY)
 /* Setting/unsetting of the history expansion character. */
 void
 sv_histchars (name)
@@ -3911,7 +3911,7 @@
       history_comment_char = '#';
     }
 }
-#endif /* BANG_HISTORY */
+#endif /* HISTORY || BANG_HISTORY */
 
 void
 sv_histtimefmt (name)
diff -urN bash-3.0-orig/variables.h bash-3.0/variables.h
--- bash-3.0-orig/variables.h	2003-07-31 16:26:47.000000000 +0200
+++ bash-3.0/variables.h	2006-05-26 09:00:37.000000000 +0200
@@ -350,10 +350,10 @@
 extern void sv_histsize __P((char *));
 extern void sv_histignore __P((char *));
 extern void sv_history_control __P((char *));
-#  if defined (BANG_HISTORY)
-extern void sv_histchars __P((char *));
-#  endif
 extern void sv_histtimefmt __P((char *));
 #endif /* HISTORY */
+#if defined (HISTORY) || defined(BANG_HISTORY)
+extern void sv_histchars __P((char *));
+#endif /* HISTORY || BANG_HISTORY */
 
 #endif /* !_VARIABLES_H_ */
